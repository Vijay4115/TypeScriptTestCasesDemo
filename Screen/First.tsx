import * as React from "react";
import axios from "axios";
import moment from "moment";
import {
  SafeAreaView,
  View,
  Text,
  TouchableOpacity,
  Dimensions,
  TextInput,
  FlatList,
} from "react-native";

const sw = Dimensions.get("window").width;
const sh = Dimensions.get("window").height;
const test = false;
interface killer {
  navigation: any;
}
interface Ss {
  apidata: [];
  searchtxt: any;
  issearch: boolean;
  searchdata: [];
  createDate: any;
}
export default class First extends React.Component<killer, Ss, {}> {
  constructor(props: killer) {
    super(props);
    this.state = {
      apidata: [],
      searchtxt: "",
      issearch: false,
      createDate: "",
      searchdata: [],
    };
  }

  componentDidMount = () => {
    this.getdata();
  };
  componentWillUnmount = () => {
    this.getdata();
  };

  getdata = () => {
    axios
      .get("https://api.github.com/repos/facebook/react/issues")
      .then((response) => {
        this.setState({ apidata: response.data });
      });
  };
  formatdate = (date: any) => {
    let Sdate = String(date);
    return moment(Sdate, "YYYYMMDD").fromNow();
  };
  fectordata = (item: any) => {
    if (item == "Type: Bug") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#fd9b9d",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.2,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#fd9b9d", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "Component: Developer Tools") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#fac905",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.45,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#fac905", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "Status: Unconfirmed") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#d2c3f9",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.45,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#d2c3f9", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "Type: Discussion") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#97BFB4",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.3,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#97BFB4", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "React 18") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#F0BB62",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.3,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#F0BB62", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "CLA Signed") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#009DAE",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.3,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#009DAE", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "React Core Team") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#E6CCA9",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.3,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#E6CCA9", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
    if (item == "Component: DOM") {
      return (
        <View
          style={{
            borderWidth: 1,
            borderColor: "#AE4CCF",
            borderRadius: 15,
            alignItems: "center",
            width: sw * 0.3,
            marginTop: 5,
            paddingBottom: 2,
          }}
        >
          <Text style={{ color: "#AE4CCF", fontSize: 12, fontWeight: "500" }}>
            {item}
          </Text>
        </View>
      );
    }
  };

  // renderData = (item: any) => {
  //   return (
    
  //   );
  // };

  onChangeInput = () => {
    let val = this.state.searchtxt;
    let tit = this.state.apidata;
    let newsearch: any = [];

    tit.map((dt: any) => {
      if (dt.title.match(val)) {
        newsearch.push(dt);
      } else {
      }
    });

    this.setState({ issearch: true, searchdata: newsearch });
  };

  render() {
    return (
      <SafeAreaView style={{ flex: 2, backgroundColor: "#131212" }}>
        <View style={{ marginTop: sh * 0.05 }}>
          <View style={{ flexDirection: "row" }}>
            <Text
              testID="Title"
              style={{ color: "#58a6ff", fontWeight: "500", fontSize: 20 }}
              

            >
              facebook / react
            </Text>
          </View>
          
          
        </View>

        <View style={{ marginTop: sh * 0.1, flexDirection: "row" }}>
          <View style={{ flex: 1 }}>
            <TextInput
              testID="txt_search"
              style={{
                borderWidth: 1,
                borderColor: "#fff",
                borderRadius: 5,
                height: sh * 0.04,
                paddingLeft: 15,
                padding: 10,
                color: "#fff",
              }}
              autoCorrect={false}
              autoCapitalize="none"              
              placeholder="is:issue is:open "
              placeholderTextColor="grey"
              onChangeText={(val) => this.setState({ searchtxt: val })}
              value={this.state.searchtxt}
            />
          </View>

          <View
            style={{
              flex: 0.2,
              marginLeft: 5,
              borderWidth: 1,
              borderColor: "#fff",
              borderRadius: 10,
              alignItems: "center",
              justifyContent: "center",
            }}
          >
            <TouchableOpacity
              testID="btn_search"
              style={{ justifyContent: "center" }}
              onPress={() => this.onChangeInput()}
            >
              <Text 
              
              style={{ color: "#fff" }}>Search</Text>
            </TouchableOpacity>
          </View>
        </View>

        <View
          style={{
            borderWidth: 1,
            borderColor: "#fff",
            margin: 15,
            marginTop: sh * 0.05,
            borderRadius: 10,
            marginBottom: 200,
          }}
        >
          {this.state.issearch ? (
            <FlatList
              testID="flt_data_search"
              data={this.state.searchdata}
              renderItem={({ item }:any) => (
                <View
                style={{
                  borderBottomWidth: 1,
                  borderColor: "#fff",
                  paddingTop: 10,
                  paddingBottom: 20,
                  paddingLeft: 15,
                  paddingRight: 10,
                  borderRadius: 5,
                }}
              >
                <View style={{ flexDirection: "row" }}>
                  <Text
                  testID="btn_navigate"
                    onPress={() =>
                      this.props.navigation.navigate("Second", { id: item.number })
                    }
                    style={{ color: "#fff", paddingRight: 45 }}
                  >
                    {item.title}
                  </Text>
                </View>
        
                <View style={{ marginLeft: 40 }}>
                  <View
                    style={{ justifyContent: "space-between", flexDirection: "row" }}
                  >
                    {item.labels[0] ? this.fectordata(item.labels[0].name) : null}
                    {item.labels[1] ? this.fectordata(item.labels[1].name) : null}
                  </View>
                  <View
                    style={{ justifyContent: "space-between", flexDirection: "row" }}
                  >
                    {item.labels[2] ? this.fectordata(item.labels[2].name) : null}
                    {item.labels[3] ? this.fectordata(item.labels[3].name) : null}
                  </View>
                </View>
                <View style={{ marginLeft: 40, marginTop: 10, flexDirection: "row" }}>
                  <Text style={{ color: "#fff" }}>#{item.number} opened</Text>
                  <Text style={{ color: "#fff", marginLeft: 10 }}>
                    {this.formatdate(item.created_at)}
                  </Text>
                  <Text style={{ color: "#fff", marginLeft: 10 }}>
                    by {item.user.login}
                  </Text>
                </View>
              </View>
              )}
            />
          ) : (
            <FlatList
              testID="flt_data"
              data={this.state.apidata}
              renderItem={({ item }:any) => (
                <View
                style={{
                  borderBottomWidth: 1,
                  borderColor: "#fff",
                  paddingTop: 10,
                  paddingBottom: 20,
                  paddingLeft: 15,
                  paddingRight: 10,
                  borderRadius: 5,
                }}
              >
                <View style={{ flexDirection: "row" }}>
                  <Text
                  testID="btn_navigate"
                    onPress={() =>
                      this.props.navigation.navigate("Second", { id: item.number })
                    }
                    style={{ color: "#fff", paddingRight: 45 }}
                  >
                    {item.title}
                  </Text>
                </View>
        
                <View style={{ marginLeft: 40 }}>
                  <View
                    style={{ justifyContent: "space-between", flexDirection: "row" }}
                  >
                    {item.labels[0] ? this.fectordata(item.labels[0].name) : null}
                    {item.labels[1] ? this.fectordata(item.labels[1].name) : null}
                  </View>
                  <View
                    style={{ justifyContent: "space-between", flexDirection: "row" }}
                  >
                    {item.labels[2] ? this.fectordata(item.labels[2].name) : null}
                    {item.labels[3] ? this.fectordata(item.labels[3].name) : null}
                  </View>
                </View>
                <View style={{ marginLeft: 40, marginTop: 10, flexDirection: "row" }}>
                  <Text style={{ color: "#fff" }}>#{item.number} opened</Text>
                  <Text style={{ color: "#fff", marginLeft: 10 }}>
                    {this.formatdate(item.created_at)}
                  </Text>
                  <Text style={{ color: "#fff", marginLeft: 10 }}>
                    by {item.user.login}
                  </Text>
                </View>
              </View>
              )}
            />
          )}
        </View>
      </SafeAreaView>
    );
  }
}
