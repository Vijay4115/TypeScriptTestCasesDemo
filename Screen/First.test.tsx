import axios from "axios";
import * as  React from "react";
import {create,act} from "react-test-renderer";
import First from "./First";
jest.useFakeTimers()


// const navigation ={
//     navigate:jest.fn()
// }
// const tree = create(<First navigation={navigation} />)

// test('Navigate To detail Screen', () => {
//         const btn = tree.root.findByProps({testID:"btn_navigate"}).props;
//         btn.onPress();
    
//         expect(navigation.navigate).toBeCalledWith('Second');
//     })
    
const tree = create(<First navigation={undefined} />)

test("Component  Check",()=>{
    let txtinput = tree.root.findByProps({testID:"txt_search"}).props;
    let Title = tree.root.findByProps({testID:"Title"}).props;
    let flt_data = tree.root.findByProps({testID:"flt_data"}).props;
    let btn_search = tree.root.findByProps({testID:"btn_search"}).props;

    expect(txtinput).toBeDefined();
    expect(Title).toBeDefined();
    expect(flt_data).toBeDefined();
    expect(btn_search).toBeDefined();
})

test("Title Test",()=>{
    let titu = tree.root.findByProps({testID:"Title"}).props;
    expect(titu.children).toBe("facebook / react")
    
})

test("First Api testing",async()=>{
    await axios
      .get("https://api.github.com/repos/facebook/react/issues")
      .then((response) => {
          
            expect(response.data).not.toBeNull();
        
      });
      
})

test("Second Api testing",async()=>{
    await axios
      .get("https://api.github.com/repos/facebook/react/issues/22879")
      .then((response) => {
        
            expect(response.data).not.toBeNull();
        
      }).catch((e)=>console.log(e)
      )
      
      
      
})

test("Search Data Test",()=>{

    let txtinput = tree.root.findByProps({testID:"txt_search"}).props;
    let btn_search = tree.root.findByProps({testID:"btn_search"}).props;
    txtinput.value  = "Fea";
    act(()=>btn_search.onPress())
    let flt_data = tree.root.findByProps({testID:"flt_data_search"}).props;
    expect(flt_data.data).not.toBeNull();
    console.log(flt_data.children);
    
    
})

test("Second Page Data of ID:#22879",async()=>{

    await axios
    .get("https://api.github.com/repos/facebook/react/issues/22879")
    .then((response) => {
      
          expect(response.data.user.login).toBe("zposten");
      
    });
})