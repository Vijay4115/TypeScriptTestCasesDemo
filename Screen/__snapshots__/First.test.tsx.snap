// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Macth SnapShot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#131212",
      "flex": 2,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 66.7,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#58a6ff",
            "fontSize": 20,
            "fontWeight": "500",
          }
        }
        testID="Title"
      >
        facebook / react
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 133.4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="is:issue is:open "
        placeholderTextColor="grey"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "#fff",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#fff",
            "height": 53.36,
            "padding": 10,
            "paddingLeft": 15,
          }
        }
        testID="txt_search"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderWidth": 1,
          "flex": 0.2,
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "opacity": 1,
          }
        }
        testID="btn_search"
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Search
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 200,
        "marginTop": 66.7,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="flt_data"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
